controllers/awsmachinepool_controller.go

```diff
@@ -117,96 +115,64 @@ func (r *AWSMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Reque
-				return ctrl.Result{}, errors.WithStack(err)
-			}
+	// remove finalizer from AWSMachinePool
+	if controllerutil.ContainsFinalizer(awsMachinePool, key.FinalizerName(iam.NodesRole)) {
+		patchHelper, err := patch.NewHelper(awsMachinePool, r.Client)
+		if err != nil {
+			return ctrl.Result{}, errors.WithStack(err)
 		}
-
-		// remove finalizer from AWSCluster
-		{
-			awsCluster, err := key.GetAWSClusterByName(ctx, r.Client, clusterName, awsMachinePool.GetNamespace())
-			if err != nil {
-				logger.Error(err, "failed to get awsCluster")
-				return ctrl.Result{}, errors.WithStack(err)
-			}
-			if controllerutil.ContainsFinalizer(awsCluster, key.FinalizerName(iam.NodesRole)) {
-				patchHelper, err := patch.NewHelper(awsCluster, r.Client)
-				if err != nil {
-					return ctrl.Result{}, errors.WithStack(err)
-				}
-				controllerutil.RemoveFinalizer(awsCluster, key.FinalizerName(iam.NodesRole))
-				err = patchHelper.Patch(ctx, awsCluster)
-				if err != nil {
-					logger.Error(err, "failed to remove finalizer on AWSCluster")
-					return ctrl.Result{}, errors.WithStack(err)
-				}
-				logger.Info("successfully removed finalizer from AWSCluster", "finalizer_name", iam.NodesRole)
-			}
+		controllerutil.RemoveFinalizer(awsMachinePool, key.FinalizerName(iam.NodesRole))
+		err = patchHelper.Patch(ctx, awsMachinePool)
+		if err != nil {
+			logger.Error(err, "failed to remove finalizer from AWSMachinePool")
+			return ctrl.Result{}, errors.WithStack(err)
 		}
+		logger.Info("successfully removed finalizer from AWSMachinePool", "finalizer_name", iam.NodesRole)
+	}
 
-		// remove finalizer from AWSMachinePool
-		if controllerutil.ContainsFinalizer(awsMachinePool, key.FinalizerName(iam.NodesRole)) {
-			patchHelper, err := patch.NewHelper(awsMachinePool, r.Client)
-			if err != nil {
-				return ctrl.Result{}, errors.WithStack(err)
-			}
-			controllerutil.RemoveFinalizer(awsMachinePool, key.FinalizerName(iam.NodesRole))
-			err = patchHelper.Patch(ctx, awsMachinePool)
-			if err != nil {
-				logger.Error(err, "failed to remove finalizer from AWSMachinePool")
-				return ctrl.Result{}, errors.WithStack(err)
-			}
-			logger.Info("successfully removed finalizer from AWSMachinePool", "finalizer_name", iam.NodesRole)
-		}
-	} else {
-		// add finalizer to AWSMachinePool
-		if !controllerutil.ContainsFinalizer(awsMachinePool, key.FinalizerName(iam.NodesRole)) {
-			patchHelper, err := patch.NewHelper(awsMachinePool, r.Client)
-			if err != nil {
-				return ctrl.Result{}, errors.WithStack(err)
-			}
-			controllerutil.AddFinalizer(awsMachinePool, key.FinalizerName(iam.NodesRole))
-			err = patchHelper.Patch(ctx, awsMachinePool)
-			if err != nil {
-				logger.Error(err, "failed to add finalizer on AWSMachinePool")
-				return ctrl.Result{}, errors.WithStack(err)
-			}
-			logger.Info("successfully added finalizer to AWSMachinePool", "finalizer_name", iam.NodesRole)
-		}
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: time.Minute * 5,
```

Comment:
Why are we requeuing?

Reply:
It was like this before so I assumed its there for a purpose.